lib/elixir/lib/kernel.ex

```diff
@@ -1206,13 +1228,22 @@ defmodule Kernel do
   @doc """
   Gets the element at the zero-based `index` in `tuple`.
 
+  It raises `ArgumentError` when `tuple` is empty,
+  or when `index` is a negative number or it is out of range of the tuple elements.
```

Comment:
`elem({:a}, 5)` will raise, and tuple is not empty and index is non negative.

if you want to compact it we can omit the empty tuple thing, because any non-negative number will be out of range.